src/main.jsx

```diff
@@ -29,39 +33,67 @@ if (localStorage.VITE_PROJECT_VERSION !== VITE_PROJECT_VERSION) {
   localStorage.clear();
 }
 
-const theme = responsiveFontSizes(
-  createTheme({
-    palette: {
-      primary: {
-        main: '#adb31b',
-        contrastText: '#fff',
-      },
-      secondary: {
-        main: '#f50057',
-      },
-    },
-  }),
-);
+const useThemePrefs = createPersistedState('themePref');
 
 const queryClient = new QueryClient();
 
+function App() {
+  const prefersDarkMode = useMediaQuery('(prefers-color-scheme: dark)');
+  const [themePrefs, setThemePrefs] = useThemePrefs();
+  const [currentTheme, setCurrentTheme] = useState();
+
+  useEffect(() => {
+    if (themePrefs === "dark") {
+      setCurrentTheme('dark');
+    } else if (themePrefs === 'light') {
+      setCurrentTheme('light');
+    } else {
+      setCurrentTheme(prefersDarkMode ? 'dark' : 'light');
+      setThemePrefs('system');
+    }
+  }, [themePrefs, prefersDarkMode])
+
+  const theme = React.useMemo(
+    () =>
+      responsiveFontSizes(createTheme({
+        palette: {
+          mode: currentTheme,
+          primary: {
+            main: '#adb31b',
+            contrastText: '#fff',
+            fg: currentTheme === 'dark' ? '#ffffff' : grey[600],
+            bg: currentTheme === 'dark' ? '#000000' : '#ffffff',
+          },
+          secondary: {
+            main: '#f50057',
+            bg: currentTheme === 'dark' ? '#333333' : '#f5f5f5',
+          },
+        },
+      })),
+    [currentTheme],
+  );
+  return (
+    <React.StrictMode>
+      <QueryClientProvider client={queryClient}>
+        <StyledEngineProvider injectFirst>
+          <ThemeProvider theme={theme}>
+            <CssBaseline />
+            <Router>
+              <Switch>
+                <Route path="/tt"><TTCount /></Route>
+                <Route path={['/', '/:place', '/:place/:day']} exact><PresencePage /></Route>
+                <Route path="/archives"><ArchivePage /></Route>
+                <Route path="*">Error 404</Route>
+              </Switch>
+            </Router>
+          </ThemeProvider>
+        </StyledEngineProvider>
+      </QueryClientProvider>
+    </React.StrictMode>
+  )
+}
+
 ReactDOM.render(
-  <React.StrictMode>
-    <QueryClientProvider client={queryClient}>
-      <StyledEngineProvider injectFirst>
-        <ThemeProvider theme={theme}>
-          <CssBaseline />
-          <Router>
-            <Switch>
-              <Route path="/tt"><TTCount /></Route>
-              <Route path={['/', '/:place', '/:place/:day']} exact><PresencePage /></Route>
-              <Route path="/archives"><ArchivePage /></Route>
-              <Route path="*">Error 404</Route>
-            </Switch>
-          </Router>
-        </ThemeProvider>
-      </StyledEngineProvider>
-    </QueryClientProvider>
-  </React.StrictMode>,
+  <App />,
```

Comment:
Maybe find another component name, and also keep `React.StrictMode` at top level.